packages/@react-spectrum/list/src/ListViewItem.tsx

```diff
@@ -140,7 +141,9 @@ export function ListViewItem(props) {
           listStyles,
           'react-spectrum-ListView-row',
           {
-            'focus-ring': isFocusVisible
+            'focus-ring': isFocusVisible,
+            'round-tops': !state.selectionManager.isSelected(item.prevKey) && (state.selectionManager.focusedKey !== item.prevKey || !isFocusVisibleFn()) && hoverState.hoveredKey !== item.prevKey,
+            'round-bottoms': !state.selectionManager.isSelected(item.nextKey) && (state.selectionManager.focusedKey !== item.nextKey || !isFocusVisibleFn()) && hoverState.hoveredKey !== item.nextKey
```

Comment:
I think the focus visible behavior isn't quite right. It shouldn't be if anything on the page has keyboard modality, it should only be if focus is somewhere inside the listview. Right now if you select something and then tab away from the listview the corners aren't rounded anymore.

Reply:
fixed in https://github.com/adobe/react-spectrum/pull/3105/commits/540adf160a59baf2c2af911de722971be9f6847c